### PR TITLE
Adding BE post override

### PR DESCRIPTION
**Pull request by WeissSa**
## Pull Request Summary
Added override field which when false makes the dao check if the project_id already exists. if it does, it rejects the post.

### Change types
- [x] Breaking change (new feature which messes with old one)
- [x] Adding tests
- [x] Adding to API

### Testing
- [x] I have added testing to this PR 

### How to manually test this PR (QA instructions)
Check MongoDB to see if there are intents with project_id 1. If there aren't. Then post should work as normal. Otherwise, post our test.json (notice all project_ids are 1) and verify that you get the expected error back in the console.

## Questions
Please test this code both manually and review the algorithms, specifically for efficiency.

### CODE REVIEW

**Review by KayleeLChan — CHANGES_REQUESTED** [review]
All the tests and QAs pass as expected! Database updates properly and axios error 500 pops up into console when id exists. Code looks good and seems like it'll run efficiently.

1 small question, not too important, but noting it here in case you wanted it implemented in this pr: In the modal, it looks like there's supposed to be a web alert when the id exists/upload runs into error (since it looks like it's awaiting error 500?) but I didn't get any web alerts when uploading an existing project id. Is this meant to happen?

**Comment by WeissSa** [comment]
@KayleeLChan check the console. It sends back a status 500. I think this is an issue with modal implementation.

File: back-end/api/controllers/transcript_controller.js [hunk 1]
```diff
@@ -34,9 +34,11 @@ export default class TranscriptController {
         try {
             const body = JSON.stringify(req.body.payload);
             const transcript = JSON.parse(body);
-            await this.#inputBoundary.formatTranscript(transcript);
+            console.log(transcript)
```
Comment (helias22) [inline back-end/api/controllers/transcript_controller.js:37]: I don't think we should have console.logs in the final version of our codes. I think I also left one in the interactor which I will delete in my next PR.

File: back-end/api/controllers/transcript_controller.js [hunk 2]
```diff
@@ -34,9 +34,11 @@ export default class TranscriptController {
...
+            const override = transcript.override;
+            await this.#inputBoundary.formatTranscript(transcript, override);
             const output = this.#outputBoundary.getOutput();
-            const status = Object.keys(output)[0];
+            const status = output.status;
```
Comment (helias22) [inline back-end/api/controllers/transcript_controller.js:41]: :)

**Review by helias22 — APPROVED** [review]
LGTM :)

File: back-end/api/controllers/transcript_controller.js (hunk 1 above)
Comment (WeissSa) [reply back-end/api/controllers/transcript_controller.js:37]: oops 100%. Missed this one

**Review by emilylwan — APPROVED** [review]
Everything looks like it should be working. I haven't had a chance to manually test the code but should be good to go.

**Pull request merged**